mod/solr_api/start.php

```diff
@@ -0,0 +1,302 @@
+		// guid must be an integer
+		if (is_numeric($guid)) 
+			$ids[] = $guid;
+		else
+			return "there is an issue deleting a record in the database, please see function delete_updated_index_list()";
+	}
+
+	if (sizeof($ids) > 0) {
+		$ids = implode(",", $ids);
+		$query = "DELETE FROM deleted_object_tracker WHERE id IN ({$ids})";
+		$result = delete_data($query);
+	}
+
+	$arr[] = array('message' => 'deleted the following guid');
+	$arr[] = $ids;
+
+	return $arr;
+}
+
+// this is not an api call, this is a hook that will initiate once user attempts deletion
+function intercept_object_deletion($event, $type, $object) {
+	$unixtime = time();
+
+	$query = "INSERT INTO deleted_object_tracker (id, time_deleted) VALUES ({$object->guid}, {$unixtime})";
+
+	// just in case for some reason, the user is able to delete the same thing twice
+	try {
+		$insert_record = insert_data($query);
+	} catch (Exception $e) {
+		error_log("error occurred: {$e}");
+	}
+
+	$query = "SELECT * FROM deleted_object_tracker";
+	$deleted_records = get_data($query);
+
+	foreach ($deleted_records as $deleted_record) {
+		error_log("the id: " . $deleted_record->id);
+	}
+
+
+	return false;
+}
+
+
+// api calls with sample set of limit 15
+function get_list_of_deleted_records() {
+
+	$query = "SELECT * FROM deleted_object_tracker";
+	$deleted_records = get_data($query);
+
+	foreach ($deleted_records as $deleted_record) {
+		error_log("the id: " . $deleted_record->id);
+		//$datetime = new DateTime("@{$deleted_record->time_deleted}");
+		//$datetime->setTimeZone(new DateTimeZone('America/New York'));
+		$arr[] = array(
+			'guid' => $deleted_record->id,
+			'time_deleted' => $deleted_record->time_deleted
+		);
+	}
+	return $arr;
+}
+
+
+function get_user_list($offset) {
+
+	$site_url = elgg_get_site_url();
+	$query = "	SELECT e.guid, ue.name, ue.username, ue.email, e.type, e.time_created, e.enabled
+				FROM elggusers_entity ue 
+					LEFT JOIN elggentities e ON ue.guid = e.guid 
+				WHERE e.enabled = 'yes'
+				LIMIT 10 OFFSET {$offset}";
+
+	$users = get_data($query);
+
+	foreach ($users as $user) {
+		$arr[] = array (
+			'guid' => $user->guid,
+			'name' => array('en' => $user->name, 'fr' => $user->name),
+			'username' => $user->username,
+			'email' => $user->email,
+			'type' => $user->type,
+			'date_created' => date("Y-m-d\TH:m:s\Z", $user->time_created),
+			'date_modified' => date("Y-m-d\TH:m:s\Z", $user->time_created),
+			'url' => "{$site_url}profile/{$user->username}"
+		);
+	}
+    return $arr;
+}
+
+
+function get_group_list($offset) {
+
+	$groups = elgg_get_entities(array(
+		'type' => 'group',
+		'limit' => 10,
+		'offset' => $offset
+	));
+
+	foreach ($groups as $group) {
+
+		if (is_Json($group->name)) {
+			$name_array = json_decode($group->name, true);
+			$name_array['en'] = str_replace('"', '\"', $name_array['en']);
+			$name_array['fr'] = str_replace('"', '\"', $name_array['fr']);
+		} else {
+			$name_array['en'] = $group->name_array;
```

Comment:
$group->name_array? Also, gc_explode_translation( $string, $lang ) does most of this already, you can just apply the replace you use to the output of that if needed.

Reply:
this was questionable because i was thinking that the general public can use this plugin for their solr instance (or for whatever reason they want with all the entities in elgg) that function call is either defined in wet4 or gcconnex_theme? and those plugins are tailored for our requirements

Reply:
Yeah, it's in wet4. The although the en/fr content json string is also part of our modifications.